frontend/src/providers/authProvider/index.tsx

```diff
@@ -10,7 +9,7 @@ import {
 } from '../../interfaces/types';
 
 const defaultValue: ContextTypes = {
-  user: {},
+  user: {} as UserType,
```

Comment:
Why {} as? Why not just "UserType"?

Reply:
This is an object with default values, not type definition.
I made these changes in consultation with @pavlovic-nemanja, typescript wouldn't let us access user.role in the router without this change.

Reply:
Why not null then?

Reply:
Yes, that's better